pandas/core/resample.py

```diff
@@ -1085,7 +1086,23 @@ def interpolate(
         """
         assert downcast is lib.no_default  # just checking coverage
         result = self._upsample("asfreq")
-        return result.interpolate(
+
+        # If the original data has timestamps which are not aligned with the
+        # target timestamps, we need to add those points back to the data frame
+        # that is supposed to be interpolated. This does not work with
+        # PeriodIndex, so we skip this case.
```

Comment:
can you add "GH#21351" to this comment somewhere

Reply:
Yes